AnnotationModule.py

```diff
@@ -0,0 +1,63 @@
+import csv
+import xml.etree.ElementTree as ET
+import cv2
+import numpy as np
+import openslide
+import os
+from skimage import io
+from BaseImage import printMaskHelper
+from PIL import Image, ImageDraw
+
+def getPoints(xmlfile):
+    # create element tree object
+    tree = ET.parse(xmlfile)
+
+    # get root element
+    root = tree.getroot()
+    annCount = 0
+
+    # create empty list for annotations
+    annotations = []
+
+    for annotation in root.findall('Annotation'):
+        for regions in annotation.findall('Regions'):
+            for region in regions.findall('Region'):
+                for vertices in region.findall('Vertices'):
+                    points[annCount] = [None] * len(vertices.findall('Vertex'))
+                    k = 0
+                    for vertex in vertices.findall('Vertex'):
+                        points[annCount][k] = (int(float(vertex.get('X'))), int(float(vertex.get('Y'))))
+                        k = k + 1
+                annCount = annCount + 1
+
+    return points
+
+def maskOutAnnotation(s,params):
+
+    annotationPath = params.get("annotation_path",s["dir"])
+    annotationPath = annotationPath + os.sep + s["filename"].rsplit(".",1)[0] + ".xml"
+
+    points = getPoints(annotationPath)
+
+    downsize_factor = int(s["height"])/s["img_mask_use"].shape[0]
+
+    points = [(a[0][0] / downsize_factor, a[0][1] / downsize_factor) for a in zip(points)]
+
+    img = Image.new('L', (s["img_mask_use"].shape[1], s["img_mask_use"].shape[0]), 0)
+
+    for pointSet in points:
+        ImageDraw.Draw(img).polygon(pointSet, outline=1, fill=1)
+
+    mask = np.array(img)
+    io.imsave(s["outdir"] + os.sep + s["filename"] + "_annotation.png", mask * 255)

```

Comment:
do you want to save the annotation mask, or the difference between the previous img_make_use and the new img_mask_use or both?

Reply:
I think I want to save the annotation mask. Does it make sense to do that?

Reply:
its up to you : ) whatever you think is most useful. you can also opt to save the fused version of the two masks. if you're happy with the results, then likely others will be, but given the open source nature of histoqc we'll be able to get feedback if they're not